rpcserver.go

```diff
@@ -567,6 +569,17 @@ func MainRPCServerPermissions() map[string][]bakery.Op {
 	}
 }
 
+// AuxDataParser is an interface that is used to parse auxiliary custom data
+// within RPC messages. This is used to transform binary blobs to human-readable
+// JSON representations.
+type AuxDataParser interface {
+	// InlineParseCustomData replaces any custom data binary blob in the
+	// given RPC message with its corresponding JSON formatted data. This
+	// transforms the binary (likely TLV encoded) data to a human-readable
+	// JSON representation (still as byte slice).
+	InlineParseCustomData(msg proto.Message) error
+}
```

Comment:
This commit adds an interface without an implementation. Was that intentional? I think we should clarify the commit message if it was intentional. Otherwise the implementation is unexpectedly missing.

Reply:
Yes, everything in the `AuxComponents` is optional (in the sense that it doesn't **have** to be provided), which is enforced by the use of the `fn.Option()`. But I added a simple unit test to demonstrate its use and updated the commit message.